src/main/io/vtx.c

```diff
@@ -73,37 +75,45 @@ void vtxProcess(timeUs_t currentTimeUs)
 {
     static timeUs_t lastCycleTimeUs;
     static uint8_t scheduleIndex;
+    bool powerUpdatePending = false;
 
     if (vtxCommonDeviceRegistered()) {
         const uint8_t currentSchedule = vtxParamSchedule[scheduleIndex];
         // Process VTX changes from the parameter group at 10Hz
         if (currentTimeUs > lastCycleTimeUs + VTX_PARAM_CYCLE_TIME_US) {
             switch (currentSchedule) {
```

Comment:
Sorry to come up with this again but that switch is a bit hard to look at I think. Why not move the band/chan and power parts in dedicated methods and just deal with the "scheduling" in this function. I guess this new feature will eventually require some more complex behavior (e.g. timeout) and that might be more clear if split.

Did you look at https://github.com/lostcontrol/betaflight/commit/11724510019b512afd798092a487a06857336a1f for that part?

Reply:
@lostcontrol See the lastest commit -- I tend to agree that it was getting murky in the vtxProcess function.  This should simplify the readability a bit.

Reply:
@codecae yes, looks good. What about a timeout before applying full power? Again, I often toggle the arm switch quickly before putting my goggles on to check that props are spinning. Don't really want to switch VTX power twice within 1 second in this case. I'm sure many people do that. This should be easy to add.

Also what about CMS? I never use CLI at the field since OSD is sooo great. Would be nice to have this feature easily configurable. Or via MSP (e.g. for telemetry).

Could be good to ask in #4334 or anywhere else (slack? rcgroups.com?) how people would use this.

Reply:
We can try messing with some sort of configurable delay if it seems popular with others.  CMS is not a big deal to do, I just struggle with finding enough available real estate on the existing screens, especially for our NTSC users.

Open to suggestions.  Would be best to keep the discussions in the issue thread so we don't get stuck playing the 'phone game'

Reply:
Adding to MSP should also be pretty straight-forward.  I think it's prudent to satisfy the requirement first, though, so we know how to modify the other extensions (CMS/OSD, MSP, etc).